markdown/extensions/toc.py

```diff
@@ -136,6 +136,8 @@ def __init__(self, md, config):
             self.use_permalinks = config["permalink"]
         self.header_rgx = re.compile("[Hh][123456]")
         self.toc_depth = config["toc_depth"]
+        if type(self.toc_depth) is not int:
+            self.toc_depth = self.toc_depth.split('-')
```

Comment:
If we're going to split the range here, why not due the type conversion here as well. When we don't need all the try/except blocks later. I realize this is a different suggestion than previously, but previously I envisioned a couple lines at the top of the block of code which did the type conversion, then the actual conditions wouldn't be so complicated.

Also wherever we do the type conversion we should define `top` and `bottom` as internal vars for later use in the conditionals. That should eliminate the need for try/except blocks in the conditions.

Reply:
I implemented your suggestions and now this looks much simpler for me.
Without the try/except blocks we lose catching "bad range errors". I am good with that.